model-optimizer/extensions/ops/select_test.py

```diff
@@ -151,29 +151,38 @@ def test_select_infer_condition_shapes_broadcast(self, else_data_shape, than_dat
         self.assertTrue(flag, resp)
 
     @generate(*[
-        ([5, 6], [5, 6], [5, 6], np.ones(np.array([5, 6]), dtype=np.float),
-         np.zeros([5, 6], dtype=np.float), np.ones([5, 6], dtype=np.float),
-         np.ones([5, 6], dtype=np.float)),
-        ([15, 3, 5], [15, 1, 5], [15, 3, 5], np.ones([15, 3, 5], dtype=np.float),
-         np.zeros([15, 3, 5], dtype=np.float), np.ones([15, 3, 5], dtype=np.float),
-         np.ones([15, 3, 5], dtype=np.float)),
-        ([15, 3, 5], [15, 1, 5], [15, 3, 5], np.ones([15, 3, 5], dtype=np.float),
-         None, np.ones([15, 3, 5], dtype=np.float), np.ones([15, 3, 5], dtype=np.float)),
-        ([15, 3, 5], [15, 1, 5], [15, 3, 5], np.ones([15, 3, 5], dtype=np.float),
-         np.ones([15, 3, 5], dtype=np.float), None, None),
-        ([15, 3, 5], [15, 1, 5], [15, 3, 5], np.zeros([15, 3, 5], dtype=np.float),
-         None, np.ones([15, 3, 5], dtype=np.float), None),
-        ([15, 3, 5], [15, 1, 5], [15, 3, 5], np.zeros([15, 3, 5], dtype=np.float),
-         np.ones([15, 3, 5], dtype=np.float), None, np.ones([15, 3, 5], dtype=np.float)),
-        ([15, 3, 5], [15, 1, 5], [15, 3, 5], np.array([True], np.bool),
-         np.zeros([15, 3, 5], dtype=np.float), np.ones([15, 3, 5], dtype=np.float),
-         np.ones([15, 3, 5], dtype=np.float)),
-        ([15, 3, 5], [15, 1, 5], [15, 3, 5], np.array([False], np.bool),
-         np.zeros([15, 3, 5], dtype=np.float), np.ones([15, 3, 5], dtype=np.float),
-         np.zeros([15, 3, 5], dtype=np.float)),
+        ([5, 6], [5, 6], [5, 6], lambda x: np.ones(x, dtype=np.float),
+         lambda x: np.zeros(x, dtype=np.float), lambda x: np.ones(x, dtype=np.float),
```

Comment:
I don't see tests where the ranks of "else", "when", "condition" are different. According to the specification all three tensors must be broadcast-able to the same shape https://numpy.org/doc/stable/reference/generated/numpy.where.html

Please, add such a tests.